### PR TITLE
Regular expressions example

### PR DESCRIPTION
**Pull request by ioanarahau**

### CODE REVIEW

File: babynames.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  f = open(filename, 'r')
+  strings = re.findall(r'<td>(\d+)</td><td>(\w+)</td><td>(\w+)', f.read())
+  for tuple in strings:
+    boy_names = (tuple[1], tuple [0])
+    girl_names = (tuple[2], tuple [0])
+    names_dict.append(boy_names)
+    names_dict.append(girl_names)
+  sorted_names = sorted(names_dict)
+  for tuple in sorted_names:
+    name = tuple[0]
```
Comment (CosminTraistaru) [inline babynames.py]: you can also use:
`name, rank = tuple`
and you will have the same result

File: babynames.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+Suggested milestones for incremental development:
+ -Extract the year and print it
+ -Extract the names and rank numbers and just print them
+ -Get the names data into a dict and print it
+ -Build the [year, 'name rank', ... ] list and print it
+ -Fix main() to use the extract_names list
+"""
+
+
+def extract_names(filename):
```
Comment (CosminTraistaru) [inline babynames.py:38]: You can also add the file you used for this.

File: babynames.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  """
+  Given a file name for baby.html, returns a list starting with the year string
+  followed by the name-rank strings in alphabetical order.
+  ['2006', 'Aaliyah 91', Aaron 57', 'Abagail 895', ' ...]
+  """
+  # +++your code here+++
+  names_dict = []
```
Comment (CosminTraistaru) [inline babynames.py]: Confusing name, this is a list not a dict.

File: babynames.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  names = []
+  f = open(filename, 'r')
```
Comment (CosminTraistaru) [inline babynames.py]: To make sure you don't forget to close a file. 
You can also use 'with':
```
with open(filename, 'r') as f:
    read_data = f.read()
```
https://docs.python.org/3/tutorial/inputoutput.html#methods-of-file-objects
And also do it in a pythonic way :)

File: babynames.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    # We didn't find a year, so we'll exit with an error message.
+    sys.stderr.write('Couldn\'t find the year!\n')
+    sys.exit(1)
+  year = year.group(1)
+  names.append(year)
+  print year
+  f.close()
+  f = open(filename, 'r')
+  strings = re.findall(r'<td>(\d+)</td><td>(\w+)</td><td>(\w+)', f.read())
+  for tuple in strings:
```
Comment (CosminTraistaru) [inline babynames.py]: The name "tuple" it's a build-it type into Python. 
It is not recommended to overwrite it. 
Even though Python let's you do that.
https://docs.python.org/3/library/stdtypes.html?highlight=tuple#tuple

File: babynames.py (hunk 1 above)
Comment (ioanarahau) [reply babynames.py]: thanks. :

File: babynames.py (hunk 3 above)
Comment (ioanarahau) [reply babynames.py]: it was a dict at first :))
I will change the name

File: babynames.py (hunk 4 above)
Comment (ioanarahau) [reply babynames.py]: :) thanks

**Pull request merged**